libraries/rush-sdk/src/proxy.ts

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
```

Comment:
`proxy` is a rather ambiguous entry point name. I'd be inclined to call this file (and the associated entry in package.json exports) `loadForFolder`, since its supported use case is to load a version of rush-lib for a folder that is probably _not_ the current working directory.

Reply:
After some discussion, I'm renaming the entry point to just `loader`.  Loading "for folder" seems like not very significant information -- the main point is that this entry point exposes a secondary API used to manage how the engine gets loaded.

1. Original entry point
```ts
// The current design that tries different heuristics to load "rush-lib" automatically
import { RushConfiguration } from '@rushstack/rush-sdk';
```

2. An optimization for bundling code without bundling the loader: (in a future PR)
```ts
// An optimized entry point for bundling plugins.
// This entry point will throw an exception if someone hasn't already
// loaded "rush-sdk" beforehand:
import { RushConfiguration } from '@rushstack/rush-sdk/preloaded';
```

3. The new API:
```ts
// The new API that allows VS Code extensions to control how the SDK is loaded:
import { rushSdkLoader, ISdkCallbackEvent } from '@rushstack/rush-sdk/loader';

if (!rushSdkLoader.alreadyLoaded) {
  // Throws an exception if alreadyLoaded=true
  await rushSdkLoader.loadAsync({ 
    rushJsonSearchFolder: "path/to/my-repo/apps/my-app",
    . . .
  });
}

const rushSdk = require('@rushstack/rush-sdk');  // access the "rush-lib" engine
```

Reply:
And note that if a plugin uses `'@rushstack/rush-sdk/preloaded'`, this can work with unit tests; they just need to perform #<!---->1 or #<!---->3 **_before_** importing the plugin code.